### ISSUE TITLE
False positive no-member on member of generic parent class with overriden __class_getitem__

### ISSUE THREAD

**Issue by badbadbatman** [PyCQA/pylint#7379]
### Bug description

```python
class Base(Generic[_T]):
    foo = "test"

    def __class_getitem__(cls, item):
        return super().__class_getitem__(item)


class Derrived(Base[int]): ...


Derrived.foo
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
E1101: Class 'Derrived' has no 'foo' member (no-member)
```

### Expected behavior

No message is reported.

### Pylint version

```shell
pylint 2.14.5
astroid 2.11.7
Python 3.8.13 (default, May  8 2022, 17:53:05)

But the same behavior on latest versions too.
```

**Comment by couteau** [comment]
Another variant of this is as follows:

```python
from typing import Generic, TypeVar

_T = TypeVar('_T')

class Base(Generic[_T]):
    foo = "test"

    def __class_getitem__(cls, item):
        return super().__class_getitem__(item)


class Derrived(Base[T]):
    def do_foo(self):
        return self.foo
```

Results in: E1101: Instance of 'Derrived' has no 'foo' member (no-member)
